conan/internal/api/uploader.py

```diff
@@ -271,7 +272,11 @@ def gzopen_without_timestamps(name, fileobj, compresslevel=None):
     return t
 
 
-def compress_files(files, name, dest_dir, compresslevel=None, ref=None):
+def compress_files(files, name, dest_dir, compresslevel=None, ref=None, cache_folder=None):
+    compress_plugin = load_compress_plugin(cache_folder)
```

Comment:
The plugin cannot be loaded for the compression of every file. It should be loaded just once by the ConfigAPI likely, then used here.